Terminal.Gui/ConsoleDrivers/CursesDriver/CursesDriver.cs

```diff
@@ -379,12 +379,39 @@ MouseEvent ProcessButtonState (Curses.MouseEvent cev, MouseFlags mf)
 			};
 		}
 
+		KeyModifiers keyModifiers;
+
+		KeyModifiers MapKeyModifiers (Key key)
+		{
+			if (keyModifiers == null)
+				keyModifiers = new KeyModifiers ();
+
+			if (!keyModifiers.Shift && key.HasFlag (Key.ShiftMask))
+				keyModifiers.Shift = true;
+			if (!keyModifiers.Alt && key.HasFlag (Key.AltMask))
+				keyModifiers.Alt = true;
+			if (!keyModifiers.Ctrl && key.HasFlag (Key.CtrlMask))
+				keyModifiers.Ctrl = true;
+			//if (!keyModifiers.Capslock)
```

Comment:
Please remove old code if it is no longer valid. You either have confidence in your PR or you don't ;-)

Reply:
I was going to remove it but then I noticed that TextField needs it, for now. But in WindowsDriver this method is not really necessary, but I have to change the TextField to use the new features. I can do it if I want to but it can also happen that other Views use it but then it is a matter of changing it.

Reply:
I think it is better to keep TextField like this for now because CursesDriver does not detect the Ctrl and Shift keys directly and therefore I am not able to map properly and will break functionality. On NetDriver I haven't seriously addressed the issue yet but I will ...

Reply:
Ahh this is on `CursesDriver`. I just kept it to be the same as `WindowsDriver `but `CursesDriver `doesn't support the other features and that's why I commented. I'll delete it then.